src/status.md

```diff
@@ -0,0 +1,313 @@
+
+### Rest parameters
+
+It has not yet been attempted to implement variadic functions due to uncertainty how efficient it will be without random stack access. The risk is that doing dynamic allocations instead may introduce an unfortunate hidden cost to function calls.
+
+In the meantime, optional function arguments, which do not have such a hidden cost, may be able to help:
+
+```ts
+function handleGiven(a: i32, b: i32 = -1, c: i32 = -1): void {
+  handle(a);
+  if (~b) {
+    handle(b);
+    if (~c) {
+      handle(c);
+    }
+  }
+}
+```
+
+### Exceptions
+
+Exceptions are not yet supported and we are waiting for the Exception handling 🦄 proposal to land. It is not yet feasible to implement exceptions without the help of the proposal as throwing and catching an exception requires stack unwinding, so the following will currently crash the program with a call to `abort("message", ...)`:
+
+```ts
+function doThrow(): void {
+  throw new Error("message")
+}
+```
+
+In the meantime we recommend to do as they did in the olden days and return an error code or `null` to indicate an exception.
+
+### Promises
+
+The concept of async execution requires an underlying concept of an event loop, which browsers and Node.js have but WebAssembly does not. In the meantime, it is recommended to stick to synchronous code within WebAssembly or call back into WebAssembly when an external async operation completes.
+
+### BigInt
+
+We have favored the use of WebAssembly's native 64-bit integers over `BigInt`s and their `INTn` notation so far, since `BigInt`s can represent values > 64-bits and as such would have to be implemented as less-efficient heap allocated objects. There are certain use cases relying on representing more than 64-bits, of course, but we still have to figure out how to support both types without introducing conflicts.
+
+### Dynamicness
+
+AssemblyScript intentionally avoids very dynamic JavaScript features that cannot be compiled efficiently, like for example:
+
+* Assigning any value to any variable.
+* Compare values of incompatible types.
+* Implicitly convert from a non-string to a string without using `x.toString()`.
+* Assign a new property, that has not been statically declared, to a class or object.
+* Assign a class to a variable (e.g. `var clazz = MyClass`) since classes are static constructs without a runtime representation.
+* Patch class `.prototype`s since there are none.
+* Access `arguments` to dynamically obtain function arguments.
+* Dynamically obtain the name of a function at runtime or otherwise use reflection.
+
+Some of these restrictions, like implicit conversion to strings when concatenating with a string, may be lifted in the future, while others, like prototypes, may never be viable in ahead-of-time compilation. For instance, some features would work in an interpreter and may become efficient with a JIT compiler, yet going down that rabbit hole runs counter to WebAssembly's, and by definition AssemblyScript's, goals.
```

Comment:
Great to call out this JS section. That features like these goes against our goals of the language, and why :smile: :tada: